flink-streaming-java/src/main/java/org/apache/flink/streaming/api/transformations/ShuffleMode.java

```diff
@@ -35,5 +35,10 @@
 	 * The producer first produces its entire result and finishes.
 	 * After that, the consumer is started and may consume the data.
 	 */
-	BATCH
+	BATCH,
+
+	/**
+	 * The shuffle mode is undefined.
```

Comment:
This could be something like `This {@link ShuffleMode} leaves it up to the framework to decide the shuffle mode. The framework will pick one of {@link BATCH} or {@link PIPELINED} in the end.`.